instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/YamlParser.java

```diff
@@ -19,93 +19,233 @@
 import com.splunk.opentelemetry.javaagent.bootstrap.nocode.NocodeExpression;
 import com.splunk.opentelemetry.javaagent.bootstrap.nocode.NocodeRules;
 import io.opentelemetry.api.trace.SpanKind;
-import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
-import java.io.InputStream;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
 import org.apache.commons.jexl3.JexlExpression;
 import org.snakeyaml.engine.v2.api.Load;
 import org.snakeyaml.engine.v2.api.LoadSettings;
 
 class YamlParser {
   private static final Logger logger = Logger.getLogger(YamlParser.class.getName());
-  // FUTURE support method override selection - e.g., with classfile method signature or something
-  private static final String NOCODE_YMLFILE = "splunk.otel.instrumentation.nocode.yml.file";
-
   private final List<NocodeRules.Rule> instrumentationRules;
   private JexlEvaluator evaluator;
 
-  YamlParser(ConfigProperties config) {
-    instrumentationRules = Collections.unmodifiableList(new ArrayList<>(load(config)));
+  public static List<NocodeRules.Rule> parseFromFile(String yamlFileName) {
+    if (yamlFileName == null || yamlFileName.trim().isEmpty()) {
+      return Collections.emptyList();
+    }
+    try {
+      try (Reader reader = Files.newBufferedReader(Paths.get(yamlFileName.trim()))) {
+        return new YamlParser(reader).getInstrumentationRules();
+      }
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Can't load configured nocode yaml.", e);
+      return Collections.emptyList();
+    }
+  }
+
+  public static List<NocodeRules.Rule> parseFromString(String yaml) {
```

Comment:
this method is only for testing?

Reply:
Yes, is there a convention to follow for that?

Reply:
could make package private (if it is in the same package as tests) and add a comment that it is only used from tests